app/scripts/modules/core/pipeline/config/stages/deploy/deployStage.js

```diff
@@ -23,7 +23,15 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.deployStage', [
         {
           type: 'stageBeforeType',
           stageTypes: ['bake', 'findAmi', 'findImage'],
-          message: 'You must have a Bake or Find Image stage before any deploy stage.'
+          message: 'You must have a Bake or Find Image stage before any deploy stage.',
+          skipValidation: function(pipeline, stage) {
+            let registry = cloudProviderRegistryProvider.$get();
```

Comment:
This is gross and I am ashamed of it. Since we're in the `config` block, we don't have access to the actual service yet, only its provider, so we have to call `$get` in the function to retrieve the service.

The alternative (at least the only one I could think of) was to rewrite the stage registration to take place in the `run` block, which would involve touching another 57 files and who knows what would have broken.